winml/lib/Api/impl/TensorBase.h

```diff
@@ -104,7 +104,9 @@ struct TensorBase : TBase {
     auto session_impl = context.session.as<winrt::Windows::AI::MachineLearning::implementation::LearningModelSession>();
     auto provider = session_impl->GetExecutionProvider();
     WINML_THROW_IF_FAILED(adapter_->GetProviderMemoryInfo(provider, dml_memory.put()));
-
+    WINML_THROW_HR_IF_TRUE_MSG(WINML_ERR_INVALID_BINDING,
+                               "DmlExecutionProvider" != adapter_->GetProviderType(provider),
```

Comment:
>DmlExecutionProvider [](start = 32, length = 20)

onnxruntime::kDmlExecutionProvider

Reply:
actually, the code already in TensorBase does this pattern to detect CPU v GPU.   copy that?

    auto spSession = context.session.as<winrt::Windows::AI::MachineLearning::implementation::LearningModelSession>();
    auto spDevice = spSession->Device().as<winrt::Windows::AI::MachineLearning::implementation::LearningModelDevice>();
    if (spDevice->IsCpuDevice()) {


look at line # 214

---
In reply to: [355645725](https://github.com/microsoft/onnxruntime/pull/2589#discussion_r355645725) [](ancestors = 355645725)

Reply:
i see what you did, you actually went to the old code this was in and just brought it fwd.  that makes sense :) 

---
In reply to: [355646168](https://github.com/microsoft/onnxruntime/pull/2589#discussion_r355646168) [](ancestors = 355646168,355645725)

Reply:
Good catch! I wonder why we got it from the provider previously.